### ISSUE TITLE
Incomplete reverse-BCP47

### ISSUE THREAD

**Issue by viktoriasee** [latex3/babel#79]
There are cases where you wish to query the locale that is actually used.

### MWE
```latex
\documentclass{article}
\usepackage[nswissgerman]{babel}
\BabelEnsureInfo

\begin{document}

\makeatletter
	\getlocaleproperty\mainlang
        {\bbl@main@language}
        {identification/tag.bcp47}
\makeatother

\typeout{THE MAIN LANGUAGE IS "\mainlang"}

Lorem ipsum.

\end{document}
```
This outputs «de». Tested under MikTeX and TexLife, XeTeX, LuaHBTeX and pdfTeX. It's not specific to Swiss German it happens to other languages as well.

### Expected behaviour
Output should be «de-CH» or even «de-CH-1996» for the 1996 spelling reform.

**Comment by jbezos** [comment]
I assigned the `<language type="..">` in the CLDR to `tag.bcp47`, which is clearly misleading. The locale actually loaded should be in `tag.ini`, but it seems it's not loaded correctly. I'll fix both issues.

**Comment by jbezos** [comment]
Fixed in 3.46. Now `tag.bcp47` contains what it's name promises.
